src/machine/aot/aot.x64.c

```diff
@@ -344,55 +346,65 @@ int aot_link(AotContext* context, size_t *szp)
   |.if WIN
     |->random_memory:
     | prepcall
+    | mov64 rax, (uint64_t)srand
+    | mov rArg1d, machine->chaos_seed
+    | call rax
+    | postcall
+    | prepcall
     | shl rcx, CKB_VM_ASM_MEMORY_FRAME_SHIFTS
-    | lea rdi, machine->memory
-    | add rdi, rcx
+    | lea rsi, machine->memory
+    | add rsi, rcx
     | mov rcx, CKB_VM_ASM_MEMORY_FRAMESIZE
     |1:
     | cmp rcx, 0
     | je >2
-    | mov64 rax, (uint64_t)rand
-    | push rdi
+    | push rsi
     | push rcx
+    | mov64 rax, (uint64_t)rand
     | call rax
     | pop rcx
-    | pop rdi
-    | mov byte [rdi], al
+    | pop rsi
+    | mov byte [rsi], al
     | sub rcx, 1
-    | add rdi, 1
+    | add rsi, 1
     | jmp <1
     |2:
+    | mov64 rax, (uint64_t)rand
+    | call rax
+    | mov machine->chaos_seed, eax
     | postcall
     | ret
   |.else
     |->random_memory:
     | prepcall
     | shl rcx, CKB_VM_ASM_MEMORY_FRAME_SHIFTS
-    | lea rdi, machine->memory
-    | add rdi, rcx
+    | lea rsi, machine->memory
+    | add rsi, rcx
     | mov rcx, CKB_VM_ASM_MEMORY_FRAMESIZE
     |1:
     | cmp rcx, 0
     | je >2
     | push rdi
+    | push rsi
     | push rcx
     | mov64 rax, (uint64_t)rand_r
     | lea rArg1, machine->chaos_seed
     | call rax
     | pop rcx
+    | pop rsi
     | pop rdi
-    | mov byte [rdi], al
+    | mov byte [rsi], al
     | sub rcx, 1
-    | add rdi, 1
+    | add rsi, 1
     | jmp <1
     |2:
     | postcall
     | ret
   |.endif
   |->inited_memory:
-  | lea rdx, machine->chaos_mode
-  | mov dl, byte [rdx]
-  | cmp dl, 0
+  | lea r8, machine->chaos_mode
```

Comment:
Any particular reason for this register change?

Reply:
rdi stores the address of the machine, if I change it at this time, then in the subsequent code, `machine->chaos_seed` will returns unexpected address

Reply:
Maybe there are some other registers, like ecx or edx we can use? The problem with r8/r8d is that they are x64 registers which will require longer instruction coding, hence longer code size. I know, we are only talking 1 or 2 bytes here, but it will just be a tiny change if we do have a register available around.

Reply:
I made a mistake here, the machine exists in `rdi` not `rdx`, I think the modification here is make no sense, but i can't run test without this modifcation, there must be a bug elsewhere.